src/base/scanfoldersmodel.cpp

```diff
@@ -345,8 +345,10 @@ void ScanFoldersModel::makePersistent()
 
 void ScanFoldersModel::configure()
 {
-    QVariantHash dirs = Preferences::instance()->getScanDirs();
+    const Preferences* pref = Preferences::instance();  // dirty workaround for ~options_imp()
```

Comment:
Still, I remember why we deleted the MainWindow later. But we have created MainWindow::cleanup() as an alternative. That's where we have to cope with this. We need to iterate through the children and delete all QDialog descendants there.